homeassistant/helpers/discovery.py

```diff
@@ -144,9 +144,6 @@ async def async_load_platform(
     """Load a component and platform dynamically.
 
     Use `async_listen_platform` to register a callback for these events.
-
-    Warning: Do not await this inside a setup method to avoid a dead lock.
-    Use `hass.async_create_task(async_load_platform(..))` instead.
```

Comment:
In rest integration, we have code that awaits `async_load_platform`:
https://github.com/home-assistant/core/blob/fbc8f0a2cf6b914d3d8a6c6b6d43db39e05bde90/homeassistant/components/rest/__init__.py#L112-L125

@MartinHjelmare mentionned in https://github.com/home-assistant/core/pull/81291 that 
> I'm not sure if this is still needed or if we changed this when we made it possible to await the platform setup for config entry forwarding.

@bdraco / @emontnemery could you confirm if this docstring can be removed?
Or maybe we do need to fix the `rest` integration?

Reply:
In theory it should be safe now but deadlocks are not easy things to see without spending hours reading code so its hard to be sure.

I'd change as many as I can of them to await before being confident in removing the docstring.